csrc/scheduler/vectorize_helper.cpp

```diff
@@ -861,6 +861,50 @@ int64_t getVectorizationFactor(
   return max_vec_size;
 }
 
+int64_t getVectorizationVectorTransposeGroup(
+    SchedulerRuntimeInfo& runtime_info,
+    TensorView* reference,
+    size_t inner_most_dim,
+    const std::vector<size_t>& dims_to_merge,
+    const std::vector<TensorView*>& vec_tv,
+    int64_t max_vectorization) {
+  max_vectorization = scheduler_utils::maxVectorizationWidth(max_vectorization);
+  std::vector<IterDomain*> virtual_innermost_dim;
+  // find the virtual_innermost_dim in reference so we can later map
+  // that to individual TensorView in vec_tv.
+  for (const auto& dim : dims_to_merge) {
+    virtual_innermost_dim.insert(
```

Comment:
Why push front instead of push back? Does the order matter?

Reply:
the order matters, since in the other PR where we refactored `mergeDims`, we need to follow the logical order from inner to outer to have aligned memory access for vectorization

Reply:
if you are worried about performance with the `insert` here, let me clean it up.

Reply:
I understand the ordering does matter for `mergeDims`, but we are not calling it with this `virtual_innermost_dim`, right?

Reply:
Sorry I wasn't being clear on this.

The reason that we need to enforce a order in `mergeDims` is because the analysis here assumes the merge order. So virtual_innermost_dim need to enforce the same logical order, i.e. when projecting the virtual_innermost_dim to the rfactor domain on reference, we'll trim iterdomains off if they come in wrong order.

The reason for the insert at begin is that, `dims_to_merge` is in the opposite order that we merge to the innermost.
so we want the merged stuff to be.
`[leaf(dims_to_merge[i-1]), leaf(dims_to_merge[i-2]), ... leaf(dims_to_merge[0]), root(inner_most_dim])]`

^^^ yes, the index for `inner_most_dim` is retrieved on root domain instead of leaf domain, hence the inconsistency.

Reply:
> i.e. when projecting the virtual_innermost_dim to the rfactor domain on reference

Are you referring to `projectId`? Didn't we talk about it just recently? Isn't the ordering of `ids` supposed not to matter?

Reply:
Yes I was.

The order of `ids` does matter. The lifted restriction from the other PR is that, we don't want to restrict `ids` to be contiguous/vectorizable on the reference tensor.

I don't think I did a good job at explaining this in the original issue :stuck_out_tongue:  Nevertheless, here's the link to the original discussion: https://github.com/NVIDIA/Fuser/pull/753#discussion_r1301947487